Implemented API key in NCBI search and fetch functions

amdevine opened the pull request:
I modified `ncbi_inst_search()`, `gb_search()`, `gb_fetch_from_id_list()`, and `ncbi_taxonomy()` functions to accept a personal NCBI API key with the argument `api_key =`  when calling the functions.  

This argument defaults to the value `False`; if a key is not supplied, api_key is not included in the parameters for the API calls.

MikeTrizna commented:
Hey Amanda, thanks for adding in these changes! We definitely need to start getting this library ready for when NCBI flips the switch on the API key.

Before I accept the pull request though, I believe the more "Pythonic" way of providing a default argument to a function is to set it to *None* rather than *False*. This has to do with the kind of crazy edge case where someone might have an API key of 0 (which is the same as *False* in Python *if* statements). Can you switch these to being *None*?

Oh also, one more thing is that I slightly modified the docstring formatting 2 weeks ago so that they will show up in the ReadTheDocs. Check it out here: http://genetic-collections.readthedocs.io/en/latest/api.html. I know I haven't gotten back to documenting the other functions, but could you include a little definition for the *api_key* argument?

amdevine commented:
Okay, I changed the default values and updated the docstrings. I think my second commit is showing up in this pull request, so hopefully if everything looks good you can just pull this one?

MikeTrizna commented:
Yep, I think your latest changes kicked off a new TravisCI build, and everything looks good to go. Thanks again!

MikeTrizna commented:
Woops, I accidentally closed the request. Let's try this again...
